### PR TITLE
SOEN-57: integrating testing frameworks

### PR DESCRIPTION
**Pull request by silentswordxv**
# Description

Adding package needed + dummy tests to ensure the frameworks actually run.

~~**I'm going to directly merge this PR so I can test the workflow**~~ Turns out the thing run on the PR so I tested it there

Part of [SOEN-57](https://soen390team8.atlassian.net/browse/SOEN-57?atlOrigin=eyJpIjoiNDczMWRmNzZjZWRkNGVhZDgwMzA3Y2ZmNDRlOGYxNDUiLCJwIjoiaiJ9)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

No tests. Only ran locally.

# Screenshots
N/A

### CODE REVIEW

**Review by stevenvuong00 — APPROVED** [review]
tested locally, works nicely, good job

**Review by KelvinTrang — APPROVED** [review]
Good job!

File: frontend/__tests__/dummy.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { render } from 'enzyme';
+
+import LoginLogoutButton from '../components/LoginLogoutButton';
+
+describe("test the login/logout button", () =>{
+    it("renders the button", async () => {
+        try {
+            render(<LoginLogoutButton/>);
```
Comment (NicholasYip) [inline frontend/__tests__/dummy.test.js:9]: shouldn't this be `expect(stuff).to.have.lengthOf(1);` ?

**Review by NicholasYip — APPROVED** [review]
Looks good, left a small comment

File: frontend/__tests__/dummy.test.js (hunk 1 above)
Comment (silentswordxv) [reply frontend/__tests__/dummy.test.js:9]: In this case no, as the purpose of the test is not to test the component specifically but to test the usage of the Enzyme library (here, render(...)).

But in a 'proper' frontend test, you might be correct. I'll have to look into it.

**Pull request merged**